src/main/java/org/synyx/urlaubsverwaltung/extension/application/ApplicationEventHandlerExtension.java

```diff
@@ -171,6 +172,7 @@ private static VacationTypeDTO toVacationType(VacationType<?> vacationType) {
             .requiresApprovalToCancel(vacationType.isRequiresApprovalToCancel())
             .color(vacationType.getColor().name())
             .visibleToEveryone(vacationType.isVisibleToEveryone())
+            .translationKey(getTranslationKey(vacationType))
```

Comment:
ist das "schon immer" teil der `VacationTypeDTOs` und hatten wir das hier vergessen?

weil wenn wir hier den key rausgeben, muss bei einen update des message keys auch ein update event versendet werden, bei den konsumenten muss darauf reagiert werden etc...

wenn der name mit rausgegeben werden soll/muss, dann stelle ich mir eine struktur a la vor

```json
{
  "name": {
    "de": "deutscher text",
    "en": "english text"
  } 
}
```

siehe VacationTypeUpdatedEventDTO.

wenn wir hier jetzt den translation key hinzufügen, dann verschlimmbessern wir den aktuellen "bug" gerade eher denke ich. stand jetzt ist das feld immer `null` und kann nicht genutzt werden. wenn wir das fixen, führen wir anfangs erwähnte komplikationen ein.

Reply:
ohne kommen die übersetzungen im zielsystem nicht richtig an und können nicht zusammengeführt werden. der key wird als neutrale id verwendet, wenn sich übersetzungen ändern.
ändert sich der key, wird eine neue übersetzung im zielsystem angelegt, anstatt ein update.

Reply:
> ohne kommen die übersetzungen im zielsystem nicht richtig an

ohne und auch mit diesem key kommen Übersetzungen nicht im Zielsystem an. Weil die Übersetzungen nicht mit gesendet werden ;-)

> und können nicht zusammengeführt werden

Das ist mein Punkt. Sie dürfen nicht "zusammengeführt" werden. Die Abwesenheitsart kommt aus der Urlaubsverwaltung. Zu der (individuellen) Abwesenheitsart gehört auch der Name in allen Sprachen. Andere System nutzen die Abwesenheitart nur, sie "kennen" sie nicht.

> ändert sich der key, wird eine neue übersetzung im zielsystem angelegt

hier kann ich noch nicht folgen.

Wenn der key übertragen wird, dann müssen die anderen System wissen, dass das ein key für einen Text ist, sie müssen mitbekommen, dass sich dieser key ändert, und in diesem Fall irgendwas machen (manuell einen Eintrag in der `message.properties` hinzufügen, keine Ahnung was machen, gerne auch automatisiert, wie auch immer. Nur... was ist denn die Übersetzung? Wo kommt der Name der Abwesenheit her? -> aus der UV


siehe [VacationTypeEventHandlerExtension](https://github.com/urlaubsverwaltung/urlaubsverwaltung/blob/main/src/main/java/org/synyx/urlaubsverwaltung/extension/vacationtype/VacationTypeEventHandlerExtension.java#L68)

Reply:
ok baue es aus